autoload/lightline/gitdiff.vim

```diff
@@ -121,6 +121,11 @@ endfunction
 " get_diff_porcelain returns the output of git's word-diff as list. The header
 " of the diff is removed b/c it is not needed.
 function! s:get_diff_porcelain() abort
+  if !s:is_git_exectuable() || !s:is_inside_work_tree()
```

Comment:
This check should be in `calculate_porcelain()` at the very beginning. Otherwise `transcode_diff_porcelain()` (and others) might complain that they don't receive a List or Dict.

Reply:
Or it should return an empty list below.